packages/node/src/indexer/dictionary/v2/ethDictionaryV2.ts

```diff
@@ -0,0 +1,279 @@
+      addressArray.push(...addresses);
+    }
+  } else {
+    if (dsOptions?.address) {
+      addressArray.push(dsOptions.address.toLowerCase());
+    }
+  }
+  return addressArray;
+}
+
+function callFilterToDictionaryCondition(
+  filter: EthereumTransactionFilter,
+  dsOptions: SubqlEthereumProcessorOptions,
+): EthDictionaryTxConditions {
+  const txConditions: EthDictionaryTxConditions = {};
+  const toArray = [];
+  const fromArray = [];
+  const funcArray = [];
+
+  if (filter.from) {
+    fromArray.push(filter.from.toLowerCase());
+  }
+  const optionsAddresses = extractOptionAddresses(dsOptions);
+  if (!optionsAddresses) {
+    if (filter.to) {
+      toArray.push(filter.to.toLowerCase());
+    } else if (filter.to === null) {
+      toArray.push(null); //TODO, is this correct?
+    }
+  } else if (optionsAddresses && (filter.to || filter.to === null)) {
+    logger.warn(
+      `TransactionFilter 'to' conflict with 'address' in data source options`,
+    );
+  }
+  if (filter.function) {
+    funcArray.push(functionToSighash(filter.function));
+  }
+
+  if (toArray.length !== 0) {
+    txConditions.to = toArray;
+  }
+  if (fromArray.length !== 0) {
+    txConditions.from = fromArray;
+  }
+
+  if (funcArray.length !== 0) {
+    txConditions.function = funcArray;
+  }
+
+  return txConditions;
+}
+
+function eventFilterToDictionaryCondition(
+  filter: EthereumLogFilter,
+  dsOptions: SubqlEthereumProcessorOptions | SubqlEthereumProcessorOptions[],
+): EthDictionaryLogConditions {
+  const logConditions: EthDictionaryLogConditions = {};
+  logConditions.address = extractOptionAddresses(dsOptions);
+  if (filter.topics) {
+    for (let i = 0; i < Math.min(filter.topics.length, 4); i++) {
+      const topic = filter.topics[i];
+      if (!topic) {
+        continue;
+      }
+      const field = `topics${i}`;
+      // Initialized
+      if (!logConditions[field]) {
+        logConditions[field] = [];
+      }
+      if (topic === '!null') {
+        logConditions[field] = []; // TODO, check if !null
+      } else {
+        logConditions[field].push(eventToTopic(topic));
+      }
+    }
+  }
+  return logConditions;
+}
+
+export function buildDictionaryV2QueryEntry(
+  dataSources: GroupedEthereumProjectDs[],
+): EthDictionaryV2QueryEntry {
+  const dictionaryConditions: EthDictionaryV2QueryEntry = {
+    logs: [],
+    transactions: [],
+  };
+
+  for (const ds of dataSources) {
+    for (const handler of ds.mapping.handlers) {
+      // No filters, cant use dictionary
+      if (!handler.filter) return dictionaryConditions;
+
+      switch (handler.kind) {
+        case EthereumHandlerKind.Block:
+          return dictionaryConditions;
+        case EthereumHandlerKind.Call: {
+          const filter = handler.filter as EthereumTransactionFilter;
+          if (
+            filter.from !== undefined ||
+            filter.to !== undefined ||
+            filter.function
```

Comment:
```suggestion
            filter.function !== undefined
```
The same changes i mentioned earlier for functions should be applied here.

Because this code is very similar for v1 and v2 it could probably be generalised